doc/how_to/custom_components/reactive_html/reactive_html_widgets.md

```diff
@@ -0,0 +1,183 @@
+
+## SVG Input
+
+This example will show you have to turn an SVG into a clickable and hoverable input widget.
+
+This can for example be used to make a technical drawing interactive.
+
+```{pyodide}
+import panel as pn
+import param
+
+from panel.reactive import ReactiveHTML
+
+pn.extension()
+
+class SVGInput(ReactiveHTML):
+    value = param.String()
+
+    clicks = param.Integer()
+    click = param.String()
+    hover = param.String()
+
+    _child_config={"value": "literal"}
+
+    _template = """\
+<style>
+.pn-container {height: 100%;width: 100%;position:relative;}
+.pn-container svg {position: relative;height:100%;width:100%}
+</style>
+<div id="container" class="pn-container"
+    onclick="${script('click_handler')}" onmouseover="${script('mouseover_handler')}"
+>
+{{ value }}
+</div>
+"""
+
+    _scripts = {
+        'click_handler': """
+const name = state.event.target.getAttribute("data-name")
+if (name!=null){
+    data.click = name
+    data.clicks += 1
+}
+""",
+        'mouseover_handler': """
+console.log("mouseover")
+const name = state.event.target.getAttribute("data-name")
+data.hover = name || ""
+"""
+}
+
+SVG = """
+<svg viewBox="0 35 300 300">
+    <rect data-name="foot" x="130" y="270" width="40" height="30" fill="brown"/>
+    <polygon data-name="bottom" points="150,150 230,270 70,270" fill="#234236"/>
+    <polygon data-name="middle" points="150,110 210,210 90,210" fill="#0C5C4C"/>
+    <polygon data-name="top" points="150,70 190,150 110,150" fill="#38755B"/>
+</svg>
+"""
+
+button = SVGInput(
+    value = SVG,
+    styles={"border": "2px solid lightgray"},
+    height=400, sizing_mode="stretch_width", max_width=1000
+)
+pn.Column(button, button.param.clicks, button.param.click, button.param.hover)
+```
+
+If you want to use your own `SVG` value, you must make sure that
+
+- the `viewBox` is set on the `svg`
+- the `data-name` is set on the ´svg` child elements, that you want to make clickable and hoverable.
+
+In this example its tempting to call the `onclick` event handler `click` instead of `click_handler`.
+But if you do, the `clicks` value will be incremented twice! Javascript `_scripts` functions
+with the same name as a Python parameter are run automatically when the Python parameter value is
+changed.
+
+## Select Widget
+
+Lets see how to create a basic `Select` widget.
+
+```{pyodide}
+import panel as pn
+import param
+
+pn.extension()
+
+class Select(pn.reactive.ReactiveHTML):
+
+    options = param.List(doc="Options to choose from.")
+
+    value = param.String(doc="Current selected option")
+
+    _template = """
+    <style>
+    .pn-container {height: 100%;width: 100%;position:relative;}
+    .style {border: 2px dashed lightgray;border-radius:20px}
+    </style>
+    <select id="select" class="pn-container style" value="${value}">
+      {% for option in options %}
+      <option id="option">{{option}}</option>
+      {% endfor %}
+    </select>
+    """
+
+    _dom_events = {'select': ['change']}
```

Comment:
I don't really understand the `_dom_events`

1. What does it really do?
  - Without it the `${value}` is not updated. I believe I have created widgets that update value if the are just in the `_template`?
2. What if I had 2 `select` tags in my `_template` but I only wanted to bind to one of them? It seems to me like I lack some control?

Reply:
I believe select is based off ID, not the widget

Reply:
Thanks. That makes much more sense. Changing it to `select_id` seems to also work.

```python
import panel as pn
import param

pn.extension()

class Select(pn.reactive.ReactiveHTML):

    options = param.List(doc="Options to choose from.")

    value = param.String(doc="Current selected option")

    _template = """
    <select id="select_id" value="${value}" style="width: ${model.width}px">
      {% for option in options %}
      <option id="option">${option}</option>
      {% endfor %}
    </select>
    """

    _dom_events = {'select_id': ['change']}

select = Select(options=['A', 'B', 'C'])
select

pn.Column(select, select.param.value).servable()
```

Reply:
The `Select` example has some issues though which I've reported in #5452